modules/gdscript/gdscript_parser.cpp

```diff
@@ -2948,6 +2948,10 @@ GDScriptParser::ExpressionNode *GDScriptParser::parse_call(ExpressionNode *p_pre
 			push_error(R"(Expected expression as the function argument.)");
 		} else {
 			call->arguments.push_back(argument);
+
+			if (argument->type == Node::IDENTIFIER && current.cursor_place == GDScriptTokenizer::CURSOR_BEGINNING) {
+				completion_context.type = COMPLETION_IDENTIFIER;
+			}
```

Comment:
Just commenting what we discussed on chat:

I'm not really familiar with the GDScript parser and completion but this looks a bit like a hack here, as if it's not the right place to do this check. No other code in this function accesses `completion_context` directly as far as I can see.

This looks more like the main loop that calls various other functions to update the completion context so my feeling is that the fix should be done in one of those functions, but again I don't really know much about it, it's just a gut feeling.

It feels like this should maybe set `ct` (and prevent it from being overridden) instead of editing the created completion context?

Would be great if @vnen could confirm, but otherwise it might be fine to merge anyway and it can be refactored later.

Reply:
The problem is the argument type and cursor place should be checked, which happened after the creation of context in the `make_completion_context` function and `parse_expression`. Otherwise, the completion would be incorrect.